test/filters/test_experimental_voxel_grid.cpp

```diff
@@ -0,0 +1,290 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Point CLoud Library (PCL) - www.pointclouds.org
+ * Copyright (c) 2021-, Open Perception
+ *
+ * All rights reserved
+ */
+
+#include <pcl/filters/experimental/voxel_grid.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/test/gtest.h>
+#include <pcl/pcl_tests.h>
+#include <pcl/point_types.h>
+
+#include <cmath>
+#include <limits>
+
+using namespace pcl;
+using namespace pcl::io;
+
+PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>);
+PointCloud<PointXYZRGB>::Ptr cloud_rgb(new PointCloud<PointXYZRGB>);
+PointCloud<PointXYZRGBA>::Ptr cloud_rgba(new PointCloud<PointXYZRGBA>);
+
+const float PRECISION = Eigen::NumTraits<float>::dummy_precision() * 2;
+
+template <typename PointT>
+void
+EXPECT_POINT_EQ(const PointT& pt1, const PointT& pt2)
+{
+  EXPECT_XYZ_NEAR(pt1, pt2, PRECISION);
+}
+
+template <typename>
+void
+EXPECT_POINT_EQ(const PointXYZRGB& pt1, const PointXYZRGB& pt2)
+{
+  EXPECT_XYZ_NEAR(pt1, pt2, PRECISION);
+  EXPECT_RGB_EQ(pt1, pt2);
+}
+
+template <typename>
+void
+EXPECT_POINT_EQ(const PointXYZRGBA& pt1, const PointXYZRGBA& pt2)
+{
+  EXPECT_XYZ_NEAR(pt1, pt2, PRECISION);
+  EXPECT_RGBA_EQ(pt1, pt2);
+}
+
+template <typename PointT>
+void
+EXPECT_POINTS_EQ(PointCloud<PointT> pc1, PointCloud<PointT> pc2)
```

Comment:
Wouldn't this be expensive? Copying 2 clouds for comparing them

Reply:
I changed it to reference